### PR TITLE
subcategory filtering with query parameter

### PR DESCRIPTION
**Pull request by JulienDemarque**

### CODE REVIEW

File: web/src/presentational/Category.js [hunk 1]
```diff
@@ -33,58 +36,83 @@ const colors = {
...
+          {subcategories.map(subcategory => (
+            <div key={subcategory}>
+              <button onClick={this.handleFilter}>{subcategory}</button>
+            </div>
+          ))}
+        </Row>
+        <Row>
+          {quizzes
+            .filter(q => {
+              return q.subcategory.indexOf(parsed.filter || 'all') >= 0;
```
Comment (adrianwix) [inline web/src/presentational/Category.js]:
```suggestion
              return parsed.filter ? q.subcategory.indexOf(parsed.filter) >= 0 : true;
```
We can take out "all" subcategory. When there is nothing selected return everything

Comment (adrianwix) [reply web/src/presentational/Category.js]: Also in the data files

File: data/javascript/index.json [hunk 2]
```diff
@@ -1,20 +1,24 @@
 {
   "category": "javascript",
+  "subcategories": ["all", "arithmetic", "coercion"],
   "quizzes": [
     {
       "title": "Simple Add",
       "key": "simple-add",
-      "categories": ["beginner", "interview"]
+      "categories": ["beginner", "interview"],
+      "subcategory": "arithmetic all"
```
Comment (adrianwix) [inline data/javascript/index.json]:
```suggestion
      "subcategory": "arithmetic"
```

File: web/src/presentational/Category.js [hunk 3]
```diff
@@ -33,58 +36,83 @@ const colors = {
...
-};
+class Category extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  handleFilter = event => {
+    const { history } = this.props;
+    const currentPath = history.location.pathname;
+    history.push(`${currentPath}?filter=${event.target.innerText}`);
```
Comment (adrianwix) [inline web/src/presentational/Category.js]:
```suggestion
    const target = event.target.innerText;
    const location = target == "all" ? `${currentPath}` : `${currentPath}?filter=${target}`;
    history.push(location);
```

File: web/src/presentational/Category.js (hunk 1 above)
Comment (JulienDemarque) [reply web/src/presentational/Category.js]: mmm would q.subcategory.indexOf(parsed.filter || '') >= 0; work then?

File: data/javascript/index.json [hunk 4]
```diff
@@ -1,20 +1,24 @@
 {
   "category": "javascript",
+  "subcategories": ["all", "arithmetic", "coercion"],
   "quizzes": [
     {
       "title": "Simple Add",
       "key": "simple-add",
-      "categories": ["beginner", "interview"]
+      "categories": ["beginner", "interview"],
+      "subcategory": "arithmetic"
```
Comment (treyhuffine) [inline data/javascript/index.json]: So I think it my head the `categories` array is the same you have in mind for the `subcategory` string. We could probably combine them.

Comment (treyhuffine) [reply data/javascript/index.json]: Also I think an array may be the better choice over a string and we could use the `.includes()` method to see if the value exists in the array.

**Pull request merged**